File: backend/src/modules/configuracion/configuracion-tienda.controller.ts
import {
  Controller,
  Get,
  Put,
  Body,
  HttpCode,
  HttpStatus,
  UseGuards,
} from '@nestjs/common';
import { ApiTags, ApiOperation, ApiResponse, ApiBearerAuth } from '@nestjs/swagger';
import { JwtAuthGuard } from '@modules/auth/guards/jwt-auth.guard';
import { ConfiguracionTiendaService } from './configuracion-tienda.service';
import { ConfigurarTiendaDto } from './dto/configurar-tienda.dto';

@ApiTags('admin/configuracion')
@ApiBearerAuth()
@UseGuards(JwtAuthGuard)
@Controller('admin/configuracion')
export class ConfiguracionTiendaController {
  constructor(private readonly configuracionService: ConfiguracionTiendaService) {}

  @Get()
  @HttpCode(HttpStatus.OK)
  @ApiOperation({ summary: 'Obtener configuración de la tienda' })
  @ApiResponse({ status: 200, description: 'Configuración obtenida exitosamente' })
  @ApiResponse({ status: 401, description: 'No autenticado' })
  async obtener() {
    return this.configuracionService.obtener();
  }

  @Put()
  @HttpCode(HttpStatus.OK)
  @ApiOperation({ summary: 'Actualizar configuración de la tienda' })
  @ApiResponse({ status: 200, description: 'Configuración actualizada exitosamente' })
  @ApiResponse({ status: 401, description: 'No autenticado' })
  async actualizar(@Body() dto: ConfigurarTiendaDto) {
    const configuracion = await this.configuracionService.actualizar(dto);
    return { mensaje: 'Configuración actualizada exitosamente', configuracion };
  }
}

File: backend/src/main/java/com/sistemapedidos/modules/configuracion/ConfiguracionTiendaController.java
package com.sistemapedidos.modules.configuracion;

import com.sistemapedidos.modules.configuracion.dto.ConfiguracionTiendaResponse;
import com.sistemapedidos.modules.configuracion.dto.ConfigurarTiendaDto;
import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PutMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.Map;

@RestController
@RequestMapping("/admin/configuracion")
@RequiredArgsConstructor
public class ConfiguracionTiendaController {

    private final ConfiguracionTiendaService configuracionService;

    @GetMapping
    public ConfiguracionTiendaResponse obtener() {
        return configuracionService.obtener();
    }

    @PutMapping
    public Map<String, Object> actualizar(@Valid @RequestBody ConfigurarTiendaDto dto) {
        ConfiguracionTiendaResponse configuracion = configuracionService.actualizar(dto);
        return Map.of("mensaje", "Configuracion actualizada exitosamente", "configuracion", configuracion);
    }
}
